tools/clusters-checker/pkg/client/client.go

```diff
@@ -0,0 +1,258 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/tidwall/gjson"
+)
+
+var (
+	log = logger.GetOrCreate("clusters-checker/pkg/client")
+
+	httpStatusesForRetry = []int{429, 502, 503, 504}
+)
+
+type esClient struct {
```

Comment:
I think its a good idea to extract the client into a separate tool that can be easier reused.